### PR TITLE
[101] - Querying the Transportation Database

### PR DESCRIPTION
**Pull request by Samju200**
## Task: `What is the total number of passengers who survived?

What is the total number of passengers who did not survive?

Get the name and sex of passengers under the age of 27 that embarked at Queenstown and Cherbourg?

How many of the passengers that embarked at Southampton survived?

Get the id, name and the total number of passengers who paid a fare greater than $100 and above the age of 35 had siblings or spouses on board?`
- `Changes you made to given task`

### CODE REVIEW

File: submissions/adeyemiJuwon/database/transportationQuery.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
+USE transportation;
+
+-- total number of passengers that survived
+SELECT COUNT(*) 
+FROM accidents 
+WHERE survived = 1;
+-- answer = 290
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/database/transportationQuery.sql:7]: Your answer is incorrect

File: submissions/adeyemiJuwon/database/transportationQuery.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+-- total number of passengers that did not survive
+SELECT COUNT(*) 
+FROM accidents 
+WHERE survived = 0;
+-- answer = 424
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/database/transportationQuery.sql:13]: same here

Comment (adaugochi) [reply submissions/adeyemiJuwon/database/transportationQuery.sql:13]: I guess you didn't import your data correctly. To even confirm, the total number of records on the CSV file for survival is `891` if you sum the two results it ought to give you that total (i.e 891). But 290 + 424 = 714 instead

File: submissions/adeyemiJuwon/database/transportationQuery.sql [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+AND point_of_embarkation != 'S';
+
+-- the total number of passengers that embarked at Southhampton and survived
+SELECT COUNT(*) 
+FROM accidents 
+LEFT JOIN trips 
+ON trips.passenger_id = accidents.passenger_id 
+WHERE point_of_embarkation = 'S' 
+AND survived = 1;
+-- answer = 202
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/database/transportationQuery.sql:30]: wrong answer

File: submissions/adeyemiJuwon/database/transportationQuery.sql [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+AND number_of_siblings_spouses != 0;
+
+-- the total number of passengers who paid a fare greater than $100 and above the age of 35 with siblings or spouses on board
+SELECT COUNT(*)
+FROM passengers
+LEFT JOIN trips 
+ON passengers.id = trips.passenger_id
+WHERE trip_fare > 100 
+AND age > 35 
+AND number_of_siblings_spouses != 0;
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/database/transportationQuery.sql:48]: You didn't add the result for this

**Review by adaugochi — CHANGES_REQUESTED** [review]
Your queries are ok, but your result is not

File: submissions/adeyemiJuwon/database/transportationQuery.sql (hunk 2 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/database/transportationQuery.sql:13]: import
I was only able to import 714 data and i have try it several time but still the same.
@adaugochi

**Comment by Samju200** [comment]
@adaugochi . The cause of my incomplete importation of the data was because my sql mode is in: ONLY_FULL_GROUP_BY, STRICT_TRANS_TABLES, NO_ZERO_IN_DATE, NO_ZERO_DATE, ERROR_FOR_DIVISION_BY_ZERO, NO_AUTO_CREATE_USER, and NO_ENGINE_SUBSTITUTION. which can not accept any empty value and some age table are empty.

**Review by adaugochi — APPROVED** [review]

**Pull request merged**